src/test/java/de/retest/recheck/review/workers/SaveFilterWorkerIT.java

```diff
@@ -23,70 +24,59 @@
+	}
 
-		final SaveFilterWorker cut = new SaveFilterWorker( applier, locateNonExistentProjectFolder );
+	@Test
+	void not_existent_parent_folder_should_be_created_upon_save() throws IOException {
+		final Path nonExistentProjectFolder = base.resolve( ".retest/recheck.ignore" );
+		final SaveFilterWorker cut = new SaveFilterWorker( createApplier( nonExistentProjectFolder ) );
 
 		cut.save();
 
 		assertThat( base.resolve( ".retest" ) ).exists();
 	}
 
 	@Test
-	void if_not_in_project_context_should_use_user_home() throws IOException {
-		final RecheckIgnoreLocator locateUserHome = new RecheckIgnoreLocator() {
-			@Override
-			public Optional<Path> getProjectIgnoreFile() {
-				return Optional.empty();
-			}
-
-			@Override
-			public Path getUserIgnoreFile() {
-				return base.resolve( "recheck.ignore" );
-			}
-		};
-
-		final SaveFilterWorker cut = new SaveFilterWorker( applier, locateUserHome );
+	void persistence_should_use_given_path() throws IOException {
+		final Path irregularFile = base.resolve( "somethingelse" );
+		final SaveFilterWorker cut = new SaveFilterWorker( createApplier( irregularFile ) );
 
 		cut.save();
 
-		assertThat( base.resolve( "recheck.ignore" ) ).hasContent( "# Comment" );
+		assertThat( irregularFile ).exists();
+		assertThat( irregularFile.toFile() ).hasContent( "# Comment" );
 	}
 
 	@Test
-	void not_existent_user_home_should_be_created_upon_save() throws IOException {
-		final RecheckIgnoreLocator locateNonExistentUserHome = new RecheckIgnoreLocator() {
-			@Override
-			public Optional<Path> getProjectIgnoreFile() {
-				return Optional.empty();
-			}
-
-			@Override
-			public Path getUserIgnoreFile() {
-				return base.resolve( ".retest/recheck.ignore" );
-			}
-		};
-		final SaveFilterWorker cut = new SaveFilterWorker( applier, locateNonExistentUserHome );
+	void multiple_different_files_should_be_resolved_correctly() throws IOException {
+		final Path file1 = base.resolve( "file1" );
+		final Path file2 = base.resolve( "file2" );
+		final Path file3 = base.resolve( "file3" );
+
+		final List<Filter> filter = new ArrayList<>();
+		filter.add( new PersistentFilter( file1, new FilterPreserveLine( "# A" ) ) );
+		filter.add( new PersistentFilter( file2, new FilterPreserveLine( "# B" ) ) );
+		filter.add( new PersistentFilter( file1, new FilterPreserveLine( "# C" ) ) );
+		filter.add( new PersistentFilter( file1, new FilterPreserveLine( "# D" ) ) );
+		filter.add( new PersistentFilter( file3, new FilterPreserveLine( "# E" ) ) );
+		filter.add( new PersistentFilter( file1, new FilterPreserveLine( "# C" ) ) );
+
+		final SaveFilterWorker cut = new SaveFilterWorker( createApplier( filter ) );
 
 		cut.save();
 
-		assertThat( base.resolve( ".retest" ) ).exists();
+		assertThat( file1.toFile() ).hasContent( "# A\n# C\n# D\n# C" );
+		assertThat( file2.toFile() ).hasContent( "# B" );
+		assertThat( file3.toFile() ).hasContent( "# E" );
```

Comment:
Same here, `Path`  should work here

Reply:
Why? Here I actually want an IT... so I check the file contents.

Reply:
Yes, I meant something like:
`assertThat(file1).hasContent("...");`